File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

        mp = {}
        count = 0
        totalSum = 0
        mp[0] = 1
        for i in range(len(nums)):
            totalSum += nums[i]
            temp = totalSum - k
            count += mp.get(temp,0)
            mp[totalSum] = mp.get(totalSum,0)+1
        
        return count
        

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer> mp = new HashMap<>();
        int sum = 0;
        int count = 0;
        mp.put(0,1);

        for(int i=0;i<nums.length;i++)
        {
            sum += nums[i];
            int temp = sum - k;
            count += mp.getOrDefault(temp,0);
            mp.put(sum,mp.getOrDefault(sum,0)+1);
        }

        return count;
    }
}